src/rpc-server.ts

```diff
@@ -0,0 +1,77 @@
+import { loadSync } from '@grpc/proto-loader';
+import { GrpcObject, loadPackageDefinition, Server } from 'grpc';
+import { inject, injectable } from 'inversify';
+import { join } from 'path';
+import { ConfigManager } from './config';
+import { Item } from './entity/Item';
+import { ConfigLoader, PersistentStorage, TYPES } from './types';
+
+@injectable()
+export class RpcServer<T> {
+
+    private _queryItem: GrpcObject;
+    private _server: Server;
+
+    constructor(@inject(TYPES.PersistentStorage) private _storage: PersistentStorage<T>,
+                @inject(TYPES.ConfigLoader) private _config: ConfigLoader) {
+        const PROTO_PATH = join(__dirname, 'protos/query-item.proto');
+
+        const packageDefinition = loadSync(PROTO_PATH, {
+            defaults: true,
+            enums: String,
+            includeDirs: [join(__dirname, 'protos')],
+            keepCase: true,
+            longs: String,
+            oneofs: true
+        });
+
+        const protoDescriptor = loadPackageDefinition(packageDefinition);
+
+        this._queryItem = protoDescriptor.queryItem as GrpcObject;
+    }
+
+    public start(host: string, port: number): void {
```

Comment:
This will make the `RpcServer` hard to test as the Server class is instantiated inside this method. You can utilize the IoC container to inject a server class.

Reply:
The Server has some API, but the behavior isn't well known currently. I will investigate whether we can create a mock for this.